cekit/cli.py

```diff
@@ -236,8 +236,7 @@ def top_context(ctx):
 def prepare_params(ctx):
     main_context = top_context(ctx)
     common_params = Map(main_context.params)
-
-    params = Map({})
+    params = Map(main_context.params)
```

Comment:
This should be taken even further. We should reuse logic similar to the one in `top_context` method, a recursive way of combining all params. It should prepare a dict with all params and we should use it directly here.

Reply:
Potentially. I'd done minimal changes to cli until a second pass to deal with the config object was done. Can discuss tomorrow.

Reply:
I was thinking of removing the `top_context` method and rewrite the `prepare_params` in something like (warning, not validated/checked):

```python
def prepare_params(ctx, params=None):
    if params is None:
        params = {}

    if ctx.parent:
        prepare_params(ctx.parent, params)

    params.update(ctx.params)

    return params
```

Reply:
Additionally conversion to `Map` object should be performed.